mocks/src/main/java/com/bumptech/glide/load/engine/executor/MockGlideExecutor.java

```diff
@@ -17,12 +17,32 @@ private MockGlideExecutor() {
     // Utility class.
   }
 
-  // Public API.
+  /**
+   * Returns a new Glide executor that delegates to the provided {@code executorService}.
+   *
+   * <p>The service provided will be considered managed by Glide and will be shutdown when {@link
+   * com.bumptech.glide.Glide#tearDown()} is called. To use an executor that should not be torn down
+   * by Glide use {@link #wrapExecutor(ExecutorService)}. If you're creating a new executor for a
+   * test (e.g. an idling resource executor) then typically you should use this method.
+   */
   @SuppressWarnings("WeakerAccess")
   public static GlideExecutor newTestExecutor(ExecutorService executorService) {
     return new GlideExecutor(executorService);
   }
 
+  /**
+   * Wraps an external executor in a {@link GlideExecutor}.
+   *
+   * <p>This should only be used when the executor used in a test is externally managed and will be
+   * shutdown by the calling code or application, Glide will not shutdown the executor when {@link
+   * com.bumptech.glide.Glide#tearDown()} is called. Most users should prefer {@link
+   * #newTestExecutor(ExecutorService)}.
+   */
+  @SuppressWarnings("WeakerAccess")
+  public static GlideExecutor wrapExecutor(ExecutorService executorService) {
```

Comment:
You could do this via your own ExecutorService wrapper that just ignores shutdown? If it's not a large burden, I'd probably prefer that to expanding the wrapper and shutdown APIs

Reply:
After trying out your suggestion I learned that this change is actually fundamentally unsafe, if we allow runnables that are submitted by Glide to run after it has awaited termination of its executors then they can run and continue interacting with Glide after it subsequently clears the caches which causes exceptions interacting with locks that have been flushed. So then this pull request shouldn't be accepted, Glide needs to be able to terminate its executors before it continues with teardown.